### PR TITLE
Allow subscribing to vehicles channel by block IDs

### PR DESCRIPTION
**Pull request by lemald**
Asana ticket: [Back-end changes to allow subscribing to vehicles by block ID](https://app.asana.com/0/0/1200347448522880/f)

These are just the back-end changes so that we can merge and deploy those first. I'll open another PR with the front-end changes shortly.

### CODE REVIEW

File: test/skate_web/channels/vehicles_channel_test.exs [hunk 1]
```diff
@@ -1,51 +1,13 @@
...
   use SkateWeb.ChannelCase
   import Test.Support.Helpers
+  import Skate.Factory
   import ExUnit.CaptureLog, only: [capture_log: 1]
 
   alias Phoenix.Socket
-  alias Realtime.Vehicle
   alias SkateWeb.{UserSocket, VehiclesChannel}
 
-  @vehicle %Vehicle{
```
Comment (phildarnowsky) [inline test/skate_web/channels/vehicles_channel_test.exs:10]: that's the stuff!

File: test/skate_web/channels/vehicles_channel_test.exs [hunk 2]
```diff
@@ -73,12 +35,25 @@ defmodule SkateWeb.VehiclesChannelTest do
 
     test "subscribes to vehicles for a run ID", %{socket: socket} do
       assert {:ok, %{data: []}, %Socket{} = socket} =
-               subscribe_and_join(socket, VehiclesChannel, "vehicles:run_ids:123-4567")
+               subscribe_and_join(socket, VehiclesChannel, "vehicles:run_ids:" <> @vehicle.run_id)
+    end
+
+    test "subscribes to vehicles for a block ID", %{socket: socket} do
```
Comment (phildarnowsky) [inline test/skate_web/channels/vehicles_channel_test.exs:41]: if we're going to support multiple block IDs in this call I'd like to see a test case for that

**Review by phildarnowsky — CHANGES_REQUESTED** [review]

File: lib/realtime/server.ex [hunk 3]
```diff
@@ -46,7 +50,8 @@ defmodule Realtime.Server do
   @typep t :: %__MODULE__{
            ets: :ets.tid(),
            active_route_ids: [Route.id()],
-           active_run_ids: [Run.id()]
+           active_run_ids: [Run.id()],
+           active_block_ids: [Block.id()]
```
[the remark this answers is not in the document]
Comment (lemald) [reply lib/realtime/server.ex:54]: Good idea, [ticket](https://app.asana.com/0/1148853526253420/1200347448522883/f) has been created.

File: test/skate_web/channels/vehicles_channel_test.exs (hunk 2 above)
Comment (lemald) [reply test/skate_web/channels/vehicles_channel_test.exs:41]: I added a test to the channel and also a test to `Realtime.Server` (which tests the actual logic more directly).

**Review by phildarnowsky — APPROVED** [review]

**Pull request merged**